### PR TITLE
Support file and tidb destDBType for drainer

### PR DESCRIPTION
**Pull request by shinnosuke-okada**
### What problem does this PR solve? 
https://github.com/pingcap/tidb-operator/issues/838

### What is changed and how does it work?
drainer config template from the tidb-cluster helm chart is changed to accept tidb and file as new drainer destDBType.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
1. Deploy drainer with `destDBType = "tidb"` -> replication is functional, and checkpoint table contains tso mapping
2. Deploy drainer with `destDBType = "mysql"` -> replication is functional, and checkpoint table doesn't contain tso mapping
3. Deploy drainer with `destDBType = "pb"` -> incremental backup is functional
4. Deploy drainer with `destDBType = "file"` -> incremental backup is functional

Code changes

 - Has Helm charts change

Side effects

 - N/A

Related changes

 - N/A

### Does this PR introduce a user-facing change?:
 
 ```release-note
NONE
 ```

### CODE REVIEW

**Comment by aylei** [comment]
@weekface @tennix PTAL

File: charts/tidb-cluster/templates/config/_drainer-config.tpl [hunk 1]
```diff
@@ -83,14 +83,11 @@ port = {{ .Values.binlog.drainer.mysql.port | default 3306 }}
...
-# Compress compresses output file, like pb and sql file. Now it supports "gzip" algorithm only.
-# Values can be "gzip". Leave it empty to disable compression.
+{{- if or (eq .Values.binlog.drainer.destDBType "pb") (eq .Values.binlog.drainer.destDBType "file") }}
+# Uncomment this if you want to use file as db-type.
 [syncer.to]
-# directory to save pb file, default same as data-dir(save checkpoint file) if this is not configured.
-dir = "/data/pb"
-compression = "gzip"
+# directory to save binlog file, default same as data-dir(save checkpoint file) if this is not configured.
+dir = "data.drainer"
```
Comment (shinnosuke-okada) [inline charts/tidb-cluster/templates/config/_drainer-config.tpl]: actually, this change is not backward compatible. i copy-pasted from the latest drainer config template but we can change it back to `/data/pb` if needed.

Comment (aylei) [reply charts/tidb-cluster/templates/config/_drainer-config.tpl]: oh yes, this should be `/data/pb` because the drainer Pod mount the PV to `/data`

Comment (shinnosuke-okada) [reply charts/tidb-cluster/templates/config/_drainer-config.tpl]: good call. reverted to `/data/pb`.

**Comment by aylei** [comment]
@shinnosuke-okada Thanks!

**Pull request merged**